bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,184 @@
+    s_conn: Connection,
+    c_handshaked: bool,
+    s_handshaked: bool,
+}
+
+/// Custom callback for verifying hostnames, need it to use s2n-tls safely
+struct HostNameHandler<'a> {
+    hostname: &'a str,
+}
+impl VerifyHostNameCallback for HostNameHandler<'_> {
+    fn verify_host_name(&self, hostname: &str) -> bool {
+        self.hostname == hostname
+    }
+}
+
+impl S2nTls {
+    /// Unsafe callback for custom IO C API
+    unsafe extern "C" fn send_cb(context: *mut c_void, data: *const u8, len: u32) -> c_int {
+        let context = &mut *(context as *mut Buffer);
+        let data = core::slice::from_raw_parts(data, len as _);
+        let ans = context.write(data).unwrap() as _;
+        ans
+    }
+
+    /// Unsafe callback for custom IO C API
+    unsafe extern "C" fn recv_cb(context: *mut c_void, data: *mut u8, len: u32) -> c_int {
+        let context = &mut *(context as *mut Buffer);
+        let mut data = core::slice::from_raw_parts_mut(data, len as _);
+        context.flush().unwrap();
+        let len = context.read(&mut data).unwrap();
+        if len == 0 {
+            info!("\t[blocking]");
+            errno::set_errno(errno::Errno(libc::EWOULDBLOCK));
+            -1
+        } else {
+            info!("\t- received {len}");
+            len as _
+        }
+    }
+
+    fn create_config(mode: Mode) -> Config {
+        let mut builder = Builder::new();
+        builder.set_security_policy(&DEFAULT_TLS13).unwrap();
+
+        match mode {
+            Mode::Server => builder
+                .load_pem(
+                    read_to_bytes(SERVER_CERT_CHAIN_PATH).as_slice(),
+                    read_to_bytes(SERVER_KEY_PATH).as_slice(),
+                )
+                .unwrap(),
+            Mode::Client => builder
+                .trust_pem(read_to_bytes(CA_CERT_PATH).as_slice())
+                .unwrap()
+                .set_verify_host_callback(HostNameHandler {
+                    hostname: "localhost",
+                })
+                .unwrap(),
+        };
+
+        builder.build().unwrap()
+    }
+
+    /// Set up connections with config and custom IO
+    fn init_conn(&mut self, mode: Mode) {
+        let c_to_s_ptr = &mut self.c_to_s_buf as &mut Buffer as *mut Buffer as *mut c_void;
+        let s_to_c_ptr = &mut self.s_to_c_buf as &mut Buffer as *mut Buffer as *mut c_void;
+        let (read_ptr, write_ptr, config, conn);
+
+        match mode {
+            Mode::Client => {
+                read_ptr = s_to_c_ptr;
+                write_ptr = c_to_s_ptr;
+                config = &self.c_config;
+                conn = &mut self.c_conn;
+            }
+            Mode::Server => {
+                read_ptr = c_to_s_ptr;
+                write_ptr = s_to_c_ptr;
+                config = &self.s_config;
+                conn = &mut self.s_conn;
+            }
+        }
+
+        conn.set_blinding(Blinding::SelfService) // no blinding so benchmark time is accurate
+            .unwrap()
```

Comment:
This feels like a lot of unwrapping. It might make more sense to have this method return a result so that you can use the `?` operator, and then you'd just call `unwrap` a single time after calling this function -> `init_conn(Client).unwrap()`.